### PR TITLE
Add diagram to Cluster Architecture concept page

### PR DESCRIPTION
**Pull request by utkarsh-singh1**
Description:
Updated /docs/en/concepts/architecture/_index.md and added kubernetes-cluster-architecture.svg in /static/images/docs.

Page to Update:
https://kubernetes.io/docs/concepts/architecture/

### CODE REVIEW

**Comment by utkarsh-singh1** [comment]
Fixes #39234

**Comment by sftim** [comment]
/retitle Add diagram to Cluster Architecture concept page

File: static/images/docs/kubernetes-cluster-architecture.svg
Comment (sftim) [inline static/images/docs/kubernetes-cluster-architecture.svg:1]: The name “cloud-controller-manager” is misspelled.

File: content/en/docs/concepts/architecture/_index.md [hunk 1]
```diff
@@ -5,3 +5,4 @@ description: >
   The architectural concepts behind Kubernetes.
 ---
 
+{{< figure src="/images/docs/components-of-kubernetes.svg" alt="Components of Kubernetes" caption="The components of a Kubernetes cluster" class="diagram-large" >}}
```
[the remark this answers is not in the document]
Comment (dipankardas011) [reply content/en/docs/concepts/architecture/_index.md:8]: As seeing the changes it looks like the image already existed and what was missing is the using it. the PR resolves that but not sure what to do with newly created architecture diagram

Also found a more verbose architecture diagram 

https://github.com/kubernetes/website/blob/main/static/images/docs/architecture.svg

**Comment by dipankardas011** [comment]
As seeing the changes it looks like the image already existed and what was missing is the using it. the PR resolves that but not sure what to do with the newly created architecture diagram

Also found a more verbose architectural diagram
https://github.com/kubernetes/website/blob/main/static/images/docs/architecture.svg

**Comment by utkarsh-singh1** [comment]
Hi @dipankardas011, you can visit issue #39234 for more info.

File: content/en/docs/concepts/architecture/_index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (utkarsh-singh1) [reply content/en/docs/concepts/architecture/_index.md:8]: Hi @dipesh-rawat, as suggested [here](https://github.com/kubernetes/website/issues/39234#issuecomment-1456156797) I have initially proposed a PR that used image in kubernetes-components and also upload the new image of cluster-architecture, upon more comments I will update with new one.

**Comment by dipankardas011** [comment]
> Hi @dipankardas011, you can visit issue #39234 for more info.

okay then. Good work 👍🏻

File: static/images/docs/kubernetes-cluster-architecture.svg
Comment (stormqueen1990) [inline static/images/docs/kubernetes-cluster-architecture.svg:1]: It looks like this file is not being used. Should it be deleted from this pull request perhaps?

**Comment by utkarsh-singh1** [comment]
Hi @stormqueen1990 @sftim @dipesh-rawat, updated the _index.md with latest commit

File: static/images/docs/kubernetes-cluster-architecture.svg
Comment (utkarsh-singh1) [reply static/images/docs/kubernetes-cluster-architecture.svg:1]: Hi @stormqueen1990 , That image is been used in updated PR.

**Review by stormqueen1990 — APPROVED** [review]

**Comment by tengqm** [comment]
Some nits:

- The spelling is incorrect: kube-api-server should be kube-apiserver;
- Use either "Scheduler" or "kube-scheduler", not "scheduler".
- If we use binary name for control plane components, then we should say "kube-controller-manager" rather than "Controller Manager".
- The color coding is confusing. Why kube-proxy, cloud-controller-manager and CLOUD PROVIDER API are colored in light blue, while other k8s component and user Pods are colored in dark blue?

**Comment by dipesh-rawat** [comment]
> Things to fix:
> 
> * The spelling is incorrect: kube-api-server should be kube-apiserver;
> * Use either "Scheduler" or "kube-scheduler", not "scheduler".
> * If we use binary name for control plane components, then we should say "kube-controller-manager" rather than "Controller Manager".
> * The color coding is confusing. Why kube-proxy, cloud-controller-manager and CLOUD PROVIDER API are colored in light blue, while other k8s component and user Pods are colored in dark blue?

@utkarsh-singh1 Could you please consider incorporating the review feedback provided [above](https://github.com/kubernetes/website/pull/42272#issuecomment-1681479345)? This would help us advance this pull request.

**Comment by utkarsh-singh1** [comment]
@tengqm, Cloud-controller-manager, cloud provider API and kube-proxy were considered as useful but optional items in the components architecture that's why they are in light blue coding schemes.

**Comment by sftim** [comment]
> cloud-controller-manager, cloud provider API and kube-proxy were considered as useful but optional items in the components architecture that's why they are in light blue coding schemes.

It's helpful to explain that to a PR reviewer - but it's much more important to make that obvious to readers who are unfamiliar with Kubernetes (for example: **add a key to the drawing**). Truth be told, website approvers already know these details. We ask this kind of question to highlight that not all readers do.

**Comment by tengqm** [comment]
Some feedbacks to the diagram:

The names of the components in the diagram should be consistent.
You can either call a component by its role, i.e. "API server",
or you can call that component by its binary name, i.e. "kube-apiserver".
We are not supposed to have both "scheduler" and "kube-proxy" in the same diagram.
It is confusing.
Please be precise with the binary name. For example, there is nothing called "kube-api-server".
It is safe to call "kube-proxy" a kubernetes component, although there are solutions
to substitute that with something else in a cluster.
"cloud-controller-manager" is not a must have. It really depends on the infrastructure
that underlies your cluster.
You may want to render "pod"s in a different color.
You may want to have CRI, CSI, CNI side by side on a node, each corresponds to
the computer, storage and network aspect of the runtime environment.
I don't get the reason why "CLOUD PROVIDER API" is capitalized.
How about change that into "Cloud Providers"?

**Comment by utkarsh-singh1** [comment]
> Some feedbacks to the diagram:
> 
> The names of the components in the diagram should be consistent.
> You can either call a component by its role, i.e. "API server",
> or you can call that component by its binary name, i.e. "kube-apiserver".
> We are not supposed to have both "scheduler" and "kube-proxy" in the same diagram.
> It is confusing.
> Please be precise with the binary name. For example, there is nothing called "kube-api-server".
> It is safe to call "kube-proxy" a kubernetes component, although there are solutions
> to substitute that with something else in a cluster.
> "cloud-controller-manager" is not a must have. It really depends on the infrastructure
> that underlies your cluster.
> You may want to render "pod"s in a different color.
> You may want to have CRI, CSI, CNI side by side on a node, each corresponds to
> the computer, storage and network aspect of the runtime environment.
> I don't get the reason why "CLOUD PROVIDER API" is capitalized.
> How about change that into "Cloud Providers"?

@tengqm, 
1. For the components like kube-proxy and cloud-controller-manager , it would be better if they are represented as different colour or marked with some character (*) so that they can be marked as optional components.
2. As a Kubernetes object , representing Pod in a different color would be a good option.
3. CRI is already mentioned in the  nodes, CSI and CNI can be mentioned in Nodes specific diagram that can be proposed later.

**Comment by tengqm** [comment]
@utkarsh-singh1  `kube-proxy` can be replaced, but not removed for most use cases. Without this component, you won't have the Service abstraction. It means you have to manually expose individual Pods to the outside world. This is doable, for example, with Gateway APIs. However, I don't think the community is there already.

**Comment by sftim** [comment]
> `kube-proxy` can be replaced, but not removed for most use cases. Without this component, you won't have the Service abstraction. It means you have to manually expose individual Pods to the outside world. This is doable, for example, with Gateway APIs. However, I don't think the community is there already.

Even though many clusters still use kube-proxy, you can remove it - and people do. See https://docs.cilium.io/en/stable/network/kubernetes/kubeproxy-free/ for an example.

**Comment by utkarsh-singh1** [comment]
>  `kube-proxy` can be replaced, but not removed for most use cases. Without this component, you won't have the Service abstraction. It means you have to manually expose individual Pods to the outside world. This is doable, for example, with Gateway APIs. However, I don't think the community is there already.

Would it be better if I mark the name like `kube-proxy*` in the diagram and below mention '* marked components are optional or can be replaced.

**Comment by tengqm** [comment]
> > `kube-proxy` can be replaced, but not removed for most use cases. Without this component, you won't have the Service abstraction. It means you have to manually expose individual Pods to the outside world. This is doable, for example, with Gateway APIs. However, I don't think the community is there already.
> 
> Even though many clusters still use kube-proxy, you can remove it - and people do. See https://docs.cilium.io/en/stable/network/kubernetes/kubeproxy-free/ for an example.

Yup. This is the only exception I am aware of so far, and I don't think that is a removal. It is a replacement. The Cilium CNI can be configured to manage Service traffics without installing iptables rules.
You will remove `kube-proxy` only when 1) you have a good understanding about its role in the cluster; 2) you have found an alternative that proxies the service traffic. Instead of putting a component like "kube-proxy or equivalent" into the diagram, I'd suggest we leave it there.

**Comment by utkarsh-singh1** [comment]
Screenshot_20230928_224331

Based on the points.

**Comment by utkarsh-singh1** [comment]
Hi @tengqm, seems like this PR was closed before I can update the architecture diagram, 

- should I leave it, 
or
- Create another PR for the updated diagram, or create another commit on same branch.

> Screenshot_20230928_224331
> 
> Based on the suggestions mention.

**Pull request merged**